crates/bls-gadgets/src/smaller_than.rs

```diff
@@ -0,0 +1,217 @@
+use algebra::{Field, PrimeField};
+use r1cs_core::{ConstraintSystem, SynthesisError};
+use r1cs_std::{
+    fields::{
+        FieldGadget,
+        fp::FpGadget,
+    },
+    boolean::Boolean,
+    ToBitsGadget,
+};
+use std::marker::PhantomData;
+
+
+pub struct SmallerThanGadget<
+    ConstraintF: Field + PrimeField,
```

Comment:
```suggestion
    ConstraintF: PrimeField,
```

`PrimeField` is `Field`